crates/rpc/src/server/health.rs

```diff
@@ -0,0 +1,54 @@
+use std::pin::Pin;
+use std::task::{Context as StdContext, Poll};
+
+use tonic::body::Body;
+use tower::{Layer, Service};
+
+/// Simple health check layer that intercepts requests to root path.
+///
+/// The root path is used by load-balancers and options requests to check the health
```

Comment:
I would name this something else. Right now it sounds like its serving a `/health` response whereas its handling requests to the root path, which happens to be used by our infra layer for health checks. But that's out-of-scope here so its confusing.